src/command_line_parser.cc

```diff
@@ -2137,37 +2144,38 @@ TritonParser::SetGlobalTraceArgs(
     bool explicit_disable_trace)
 {
   for (const auto& global_setting : lparams.trace_config_map_[""]) {
+    auto user_setting = std::get<std::string>(global_setting.second);
```

Comment:
optional: With c++17, you could use structured bindings if you wanted. Something like:
```cpp
for (const auto& [setting, value] : lparams.trace_config_map_[""])
  if (setting == "rate") {
    lparams.trace_rate_ = ParseOption<int>(value);
  }
  ...
```

Reply:
right, I forgot about this luxury in C++17